src/mbgl/style/expression/distance.cpp

```diff
@@ -33,10 +34,39 @@ std::size_t getRangeSize(const IndexRange& range) {
     return range.second - range.first + 1;
 }
 
+bool isRangeSafe(const IndexRange& range, const std::size_t threshold) {
+    return (range.second >= range.first && range.second < threshold);
+}
+
+std::pair<mbgl::optional<IndexRange>, mbgl::optional<IndexRange>> splitRange(const IndexRange& range, bool isLine) {
+    if (range.first > range.second) return std::make_pair(nullopt, nullopt);
+    auto size = getRangeSize(range);
+    if (isLine) {
+        if (size == 2) {
+            return std::make_pair(range, nullopt);
+        }
+        auto size1 = size / 2;
+        IndexRange range1(range.first, range.first + size1);
+        IndexRange range2(range.first + size1, range.second);
+        return std::make_pair(std::move(range1), std::move(range2));
+    } else {
+        if (size == 1) {
+            return std::make_pair(range, nullopt);
+        }
+        auto size1 = size / 2 - 1;
```

Comment:
what if size == 0?

Reply:
Here `size`'s minimum value is 1, as at the begainning `if (range.first > range.second) return std::make_pair(nullopt, nullopt);`